src/bucket/FutureBucket.cpp

```diff
@@ -210,7 +211,18 @@ FutureBucket::resolve()
     checkState();
     assert(isLive());
     clearInputs();
+    auto timer = ElapsedTime();
+
     std::shared_ptr<Bucket> bucket = mOutputBucket.get();
+    auto elapsed = timer.elapsed<>();
+    if (elapsed.count() > 1000)
```

Comment:
as this will be used in more than one place can we instead have a scope object that performs the conditional logging?
Something like:
```c++
std::shared_ptr<Bucket> bucket;
auto SlowLOG(INFO, "Bucket") << "Resolving bucket " << hexAbbrev(bucket->getHash());
{
    bucket= mOutputBucket.get(); 
}
```